### DISCUSSION TITLE
Autolog (Sklearn -- others?) -- Analagous generation of test metrics as training metrics

### DISCUSSION THREAD

**Discussion by pdornian** [mlflow/mlflow#10234] [Ideas]
Just started exploring mlflow. I've been using autologging in an sklearn logistic regression case as a sandbox. It's really nice that f1, precision, ROC, et al are auto generated on model training. However, I'd really like to have that exact same set of metrics also generated when evaluating a test set. I know this can be specified via model.score and model.metrics, but this requires defining each of these metrics individually (while all the training metrics and charts are generated by default without individual specification). 

It would be really convenient if there was a single command that generated exactly the analagous metrics for some test set you wanted to evaluate. 

I've only played with mlflow.sklearn so far, but figure that this might apply to the other APIs as well.
